testER/src/App.jsx

```diff
@@ -357,32 +342,40 @@ const App = (props) => {
     // toggle off all tracks??
   }
 
-  return (
-    <>
-      <div id='map-container' onKeyDown={logKey} onKeyUp={logKey}>
-        {/* <a href='https://earthranger.com/'>
+  return <TrackContext.Provider value={{ displayTracks, setTracks, tracks }}>
+    <div id='map-container' onKeyDown={logKey} onKeyUp={logKey}>
+      {/* <a href='https://earthranger.com/'>
           <img src='./public/images/LogoEarthRanger.png' id='earth-ranger-logo' />
         </a> */}
-        <Legend
-          subs={subjects}
-          track={tracks}
-          subjectData={config}
-          onTrackClick={(updatedTrack) => {
-            // const newState = tracks
-            const newState = Object.assign({}, tracks)
-            newState[updatedTrack[0]] = updatedTrack[1]
-            setTracks(newState)
-            displayTracks(updatedTrack)
-          }}
-          onLocClick={(coords) => goToLoc(coords)}
-          legSub={legSub}
-          onReturnClick={(subject) => setLegSub(subject)}
-          onStoryClick={(subject) => setLegSub(subject)}
-        />
-        {/* <p id='reset' onClick={resetMap}>RESET</p> */}
-      </div>
-    </>
-  )
+      <Legend
+        subs={subjects}
+        subjectData={config}
+        onLocClick={(coords) => goToLoc(coords)}
+        legSub={legSub}
+        onReturnClick={(subject) => setLegSub(subject)}
+        onStoryClick={(subject) => setLegSub(subject)}
+      />
+      {/* <p id='reset' onClick={resetMap}>RESET</p> */}
+    </div>
+    {subjectPopups.map(({ properties, geometry }) =>
+      <Popup
```

Comment:
Render a popup for each subjectPopup entry. onClose, invoked with the popup close event is triggered, removes that subjectPopup data from the collection. Coordinates provide the position. The children provide the markup to be rendered+updated in the popup.